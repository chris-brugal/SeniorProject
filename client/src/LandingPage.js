import React, { useEffect } from "react";
import useAuth from "./useAuth";
import SpotifyWebApi from "spotify-web-api-node";

const spotifyApi = new SpotifyWebApi({
  clientId: "dd1e4b8b29004075bfe66da0d0a488f3",
});

const LandingPage = ({ code }) => {
  const accessToken = useAuth(code);

  useEffect(() => {
    //if no access token
    if (!accessToken) return;

    spotifyApi.setAccessToken(accessToken);

    //gets user detail
    spotifyApi.getMe().then((data) => {
      console.log(data);
    });
  }, [accessToken]);

  return <div>{code}</div>;
};

export default LandingPage;
