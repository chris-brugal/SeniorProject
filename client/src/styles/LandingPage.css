.typew {
  color: #1976d2;
  font-weight: 500;
  font-size: 1.5em;
}

.main-img {
  background-size: "contain";
  max-width: 80%;
  margin-top: 10px;
  max-height: 80%;
}

.main-text {
  max-width: 100%;
  height: auto;
  justify-content: center;
}

.about-icon {
  display: flex;
  align-items: center;
  flex-wrap: wrap;
  justify-content: center;
}

.card-title {
  text-align: center;
}

.card-size {
  min-height: 200px;
  justify-content: center;
}

.icon-img {
  background-size: "contain";
  max-width: 20%;
  margin-top: 10px;
  max-height: 20%;
}

.home-div {
  margin-top: 65px;
}

.button-size {
  justify-content: center;
  margin-bottom: 5px;
  text-align: center;
}

.navbar > .active {
  border-bottom: 1px white;
}

.navbarr {
  padding-bottom: 70px;
}
