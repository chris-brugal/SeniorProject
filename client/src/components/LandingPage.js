import React from "react";
import "../styles/LandingPage.css";
import Home from "./Home.js";
import GetStarted from "./GetStarted.js";
import AboutUs from "./AboutUs.js";

const LandingPage = () => {
  return (
    <div id="home" className="home-div">
      <Home />
      <AboutUs />
      <GetStarted />
    </div>
  );
};

export default LandingPage;
