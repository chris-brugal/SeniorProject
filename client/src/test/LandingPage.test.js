import React from 'react';
import LandingPage from '../components/LandingPage';
import renderer from "react-test-renderer";

describe("snapshot testing", () => {

test("snapshot testing for Landing Page component", () => {
    const comp = renderer.create(<LandingPage />).toJSON()
    expect(comp).toMatchSnapshot()
})
})