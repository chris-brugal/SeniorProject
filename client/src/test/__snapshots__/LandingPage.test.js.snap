// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot testing for Landing Page component 1`] = `
<div
  className="home-div"
  id="home"
>
  <div
    className="home-div"
    id="home"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-sc7rel-MuiGrid-root"
        justify="center"
      >
        <h2
          className="MuiTypography-root MuiTypography-h2 main-text css-155ym27-MuiTypography-root"
        >
          Welcome to Spotify Central, begin
          <div
            className="typew"
          >
            <div
              className="Typewriter"
              data-testid="typewriter-wrapper"
            />
          </div>
          <a
            onClick={[Function]}
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Get Started
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </a>
        </h2>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7 css-19mjduq-MuiGrid-root"
      >
        <img
          alt="Music Player"
          className="main-img"
          src="/images/player.jpg"
        />
      </div>
    </div>
  </div>
  <div
    className="About Us"
    id="about"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2 main-text css-lrs2p1-MuiTypography-root"
    >
      About Spotify Central
    </h2>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-1402ce4-MuiGrid-root"
      style={
        Object {
          "minHeight": "95vh",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-1rymtov-MuiGrid-root"
        justify="center"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card-size css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiBox-root css-scc2th"
          >
            <img
              alt="graph"
              className="icon-img"
              src="/images/graph.png"
            />
          </div>
          <div
            className="MuiCardHeader-root card-title css-185gdzj-MuiCardHeader-root"
          >
            <div
              className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
              >
                Statistics
              </span>
              <span
                className="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
              >
                Be able to see your statics such as your top artists and albums!
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-1rymtov-MuiGrid-root"
        justify="center"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card-size css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiBox-root css-scc2th"
          >
            <img
              alt="trail"
              className="icon-img"
              src="/images/trail.png"
            />
          </div>
          <div
            className="MuiCardHeader-root card-title css-185gdzj-MuiCardHeader-root"
          >
            <div
              className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
              >
                Generate
              </span>
              <span
                className="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
              >
                Generate perfect length playlists for when you are on the road!
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-1rymtov-MuiGrid-root"
        justify="center"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card-size css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiBox-root css-scc2th"
          >
            <img
              alt="album"
              className="icon-img"
              src="/images/album.png"
            />
          </div>
          <div
            className="MuiCardHeader-root card-title css-185gdzj-MuiCardHeader-root"
          >
            <div
              className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
              >
                Personalize
              </span>
              <span
                className="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
              >
                Be able to customize and edit your playlists much more easily!
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="home-div"
    id="GetStarted"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2 main-text css-lrs2p1-MuiTypography-root"
    >
      Get Started
    </h2>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-z4xi2a-MuiGrid-root"
      style={
        Object {
          "minHeight": "90vh",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7 css-fim4zn-MuiGrid-root"
        justify="center"
      >
        <img
          alt="join"
          className="main-img"
          height="50%"
          src="/images/city.jpg"
          width="50%"
        />
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-sc7rel-MuiGrid-root"
        justify="center"
      >
        <h2
          className="MuiTypography-root MuiTypography-h2 main-text css-1sra7t5-MuiTypography-root"
        >
          Log in and go! 
          <br />
          <a
            onClick={[Function]}
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Back to the top
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </a>
        </h2>
      </div>
    </div>
  </div>
</div>
`;
